convert azure sub id to byte string vs UTF

esell opened the pull request:
### What does this PR do?
convert azure subscription_id to byte string vs UTF

### What issues does this PR fix or reference?
[46935](https://github.com/saltstack/salt/issues/46935)

### Previous Behavior
Using the azurearm driver would throw a "subscription_id must be a str" error

### New Behavior
Error no longer is thrown

### Tests written?

No

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

esell committed:
convert azure sub id to byte string vs UTF

terminalmage commented on salt/utils/azurearm.py:
```diff
@@ -124,7 +124,7 @@ def _determine_auth(**kwargs):
     subscription_id = kwargs['subscription_id']
 
-    return credentials, subscription_id, cloud_env
+    return credentials, subscription_id.encode(), cloud_env
```
This doesn't seem correct, since the documentation for the subscription ID in the azure mgmt modules says that it's supposed to be a `str`. Encoding would give you a bytestring, meaning this would likely break on Python 3. If what we need is a `str` type, then what we should do is:

1. Revert line 127 back to its original value.
2. Change line 125 to `subscription_id = kwargs['subscription_id']`

You'd also need to import `salt.utils.stringutils`.

terminalmage replied:
Hey, sorry I just realized I mistyped in my earlier comment. It should have been "Change line 125 to `subscription_id = salt.utils.stringutils.to_str(kwargs['subscription_id'])`."

esell commented:
@terminalmage I had not updated the other files only because I thought that the one in `salt/utils` was the area that the main Azure client connection was initiated. From there, all of those other referenced files pulled from `salt/utils/azurearm.py`. The main issue, at least as it appears to me, is the act of passing the `subscription_id` value from salt over to the Azure lib(s). Depending on the path taken to fix this, I can certainly update the other files.

Thanks for the help with this. As I mentioned before, salt and python are not my world so there are things I'm likely doing wrong haha.

esell committed:
update to use to_str() instead of encode()

Signed-off-by: esell <esell@esheavyindustries.com>

esell committed:
duh, actually remove encode()

Signed-off-by: esell <esell@esheavyindustries.com>

esell commented:
Ok, I've removed the `encode()` stuff and replaced it with the salt utility method, thanks for pointing that out. Does it make sense to run through the other files you listed and repeat the same process?
